packages/forms-web-app/src/dynamic-forms/validator/conditional-required-validator.js

```diff
@@ -0,0 +1,53 @@
+const { body } = require('express-validator');
+const BaseValidator = require('./base-validator.js');
+
+/**
+ * @typedef {import('../question.js')} Question
+ */
+
+/**
+ * enforces a field is not empty when condition is satisfied
+ * @class
+ */
+class ConditionalRequiredValidator extends BaseValidator {
+	/**
+	 * @type {string} error message to display to user
+	 */
+	errorMessage = 'Provide further information';
+
+	/**
+	 * creates an instance of a ConditionalRequiredValidator
+	 * @param {string} [errorMessage] - custom error message to show on validation failure
+	 */
+	constructor(errorMessage) {
+		super();
+
+		if (errorMessage) {
+			this.errorMessage = errorMessage;
+		}
+	}
+
+	/**
+	 * validates the response body, checking the questionObj's fieldname
+	 * @param {Question} questionObj
+	 */
+	validate(questionObj) {
+		return questionObj.options.reduce((schema, option) => {
+			if (option.conditional) {
+				schema.push(
+					body(this.getConditionalFieldName(questionObj, option))
+						.if(body(questionObj.fieldName).equals(option.value))
+						.notEmpty()
+						.withMessage(this.errorMessage)
+				);
+			}
+			return schema;
+		}, []);
+	}
+
+	getConditionalFieldName(questionObj, option) {
```

Comment:
we could arguably have this method somewhere common, passing in fieldname and conditionalfieldname and it returning the correctly formatted string. This could then be called both by the validator and the options question class.